src/worker.cpp

```diff
@@ -1101,13 +1103,100 @@ void Worker::populateDbus(const types::VPDMapVariant& parsedVpdMap,
     }
 }
 
+std::string Worker::getAssetTag(const types::VPDMapVariant& i_parsedVpdMap)
+{
+    std::string l_assetTag;
+
+    try
+    {
+        // system VPD will be in IPZ format.
+        if (auto l_parsedVpdMap =
+                std::get_if<types::IPZVpdMap>(&i_parsedVpdMap))
+        {
+            auto l_itrToVsys = (*l_parsedVpdMap).find(constants::recVSYS);
```

Comment:
we can create this variable inside if condition .
As this variable is not being used outside anywhere.

Reply:
"l_itrToVsys"  is being used in the "if" condition itself.
We can't check it before creation.